src/main/java/com/minecolonies/core/entity/mobs/aitasks/RaiderWalkAI.java

```diff
@@ -87,43 +87,23 @@ private boolean walk()
 
                 final List<BlockPos> wayPoints = ((IColonyRaidEvent) event).getWayPoints();
                 final BlockPos moveToPos = ShipBasedRaiderUtils.chooseWaypointFor(wayPoints, raider.blockPosition(), targetBlock);
-                raider.getNavigation().moveToXYZ(moveToPos.getX(), moveToPos.getY(), moveToPos.getZ(), !moveToPos.equals(targetBlock) && moveToPos.distManhattan(wayPoints.get(0)) > 50 ? 1.8 : 1.1);
-                walkInBuildingState = false;
+                EntityNavigationUtils.walkToPos(raider, moveToPos, 4, false, !moveToPos.equals(targetBlock) && moveToPos.distManhattan(wayPoints.get(0)) > 50 ? 1.8 : 1.1);
+                walkInBuilding = null;
                 randomPathResult = null;
             }
-            else if (walkInBuildingState)
+            else if (walkInBuilding != null)
             {
-                final BlockPos moveToPos = findRandomPositionToWalkTo();
-                if (moveToPos != null)
+                if (EntityNavigationUtils.walkToRandomPosWithin(raider, 10, 0.7, walkInBuilding.getCorners())
```

Comment:
Does this take the
randomPathResult.getJob().getPathingOptions().withCanEnterDoors(true).withToggleCost(0).withNonLadderClimbableCost(0);

Special costs in account?